pytest_fixtures/certificate.py

```diff
@@ -0,0 +1,150 @@
+            connection.run('mkdir ssl-build/{}'.format(_cert_data['capsule_hostname']))
+            # Ignore creation error, but assert directory exists
+            assert connection.run('test -e ssl-build/{}'.format(_cert_data['capsule_hostname']))
+        upload_file(
+            local_file=get_data_file('generate-ca.sh'),
+            remote_file='generate-ca.sh',
+            hostname=hostname,
+        )
+        upload_file(
+            local_file=get_data_file('generate-crt.sh'),
+            remote_file='generate-crt.sh',
+            hostname=hostname,
+        )
+        upload_file(
+            local_file=get_data_file('openssl.cnf'), remote_file='openssl.cnf', hostname=hostname
+        )
+        # create the CA cert.
+        with get_connection(timeout=300, hostname=hostname) as connection:
+            result = connection.run('echo 100001 > serial')
+            result = connection.run('bash generate-ca.sh')
+            assert result.return_code == 0
+        # create the Satellite's cert
+        with get_connection(timeout=300, hostname=hostname) as connection:
+            result = connection.run(
+                'yes | bash {} {}'.format('generate-crt.sh', _cert_data['sat6_hostname'])
+            )
+            assert result.return_code == 0
+
+    return _cert_setup
+
+
+@pytest.fixture()
+def certs_cleanup():
+    """cleanup all cert configuration files"""
+    yield
+    files = [
+        'cacert.crt',
+        'certindex*',
+        'generate-*.sh',
+        'capsule_cert' 'openssl.cnf',
+        'private',
+        'serial*',
+        'certs/*',
+        settings.server.hostname,
+    ]
+    with get_connection(timeout=300) as connection:
+        files = ' '.join(files)
+        result = connection.run(f'rm -rf {files}')
+        assert result.return_code == 0
+
+
+@pytest.fixture()
+def generate_certs():
+    """generate custom certs in satellite host"""
+    upload_file(
+        local_file=get_data_file('certs.sh'),
+        remote_file='certs.sh',
+    )
+    upload_file(
+        local_file=get_data_file('extensions.txt'),
+        remote_file='extensions.txt',
+    )
+    with get_connection(timeout=300) as connection:
+        result = connection.run('bash certs.sh')
+        assert result.return_code == 0
+
+
+@pytest.fixture(scope='module')
+def file_setup():
+    """Create working directory and file."""
+    capsule_hostname = 'capsule.example.com'
+    tmp_dir = '/var/tmp/{}'.format(gen_string('alpha', 6))
+    caps_cert_file = f'{tmp_dir}/ssl-build/capsule.example.com/cert-data'
+    # Use same path locally as on remote for storing files
+    Path(f'{tmp_dir}/ssl-build/capsule.example.com/').mkdir(parents=True, exist_ok=True)
+    with get_connection(timeout=200) as connection:
+        result = ssh.command(f'mkdir {tmp_dir}')
+        assert result.return_code == 0, 'Create working directory failed.'
+        # Generate a Capsule cert for capsule.example.com
+        result = connection.run(
```

Comment:
Is there any reason you can't use (maybe with slight updates so certs_tar can be passed) `Satellite.capsule_certs_generate` for this fixture?